src/sb_handle/kernel_constructor.hpp

```diff
@@ -237,6 +235,9 @@ static SYCL_BLAS_INLINE cl::sycl::event execute_tree(queue_t q_,
   cl::sycl::event ev;
   try {
     auto cg1 = [=](cl::sycl::handler &h) mutable {
+#ifdef SB_ENABLE_USM
```

Comment:
If we are adding the event dependency also for the kernels that use buffer, it would be better to remove this pragma.

Reply:
This is just to keep the call to `depends_on` for `DPC++`. (USM and Buffers)

Reply:
Yes, but I think we are introducing the dependencies also for ComputeCPP in case of buffer.

Reply:
Yes, but that's only at the api level. We don't use those dependencies anywhere in the code.